twoliter/src/lock.rs

```diff
@@ -430,11 +404,11 @@ impl Lock {
         create_dir_all(&cache_path).await?;
 
         // First get the manifest for the specific requested architecture
-        let manifest = self.get_manifest(image, arch).await?;
-        let oci_archive = OCIArchive::new(image, manifest.digest.as_str(), &cache_path);
+        let manifest = self.get_manifest(image_tool.clone(), image, arch).await?;
+        let oci_archive = OCIArchive::new(image, manifest.digest.as_str(), &cache_path)?;
 
         // Checks for the saved image locally, or else pulls and saves it
-        oci_archive.pull_image().await?;
+        oci_archive.pull_image(image_tool.clone()).await?;
```

Comment:
I am surprised and a bit suspicious that we're not passing `arch` here. Both `docker` and `crane` allow the use of a `--platform` argument when pulling.

Reply:
Yes that is true but if we have the image digest (which at this point we do) the platform is not needed as the image digest here refers only to a specific architecture image.